create_structure/resource/hook/create_structure.py

```diff
@@ -0,0 +1,388 @@
+
+            if (
+                alias_for and isinstance(alias_for, dict) and
+                alias_for['id'].lower() == entity_type and
+                object_typeid == alias_for.get(
+                    'classifiers', {}
+                ).get('object_typeid')
+            ):
+
+                return schema['id']
+
+        for schema in self._session.schemas:
+            alias_for = schema.get('alias_for')
+
+            if (
+                alias_for and isinstance(alias_for, basestring) and
+                alias_for.lower() == entity_type
+            ):
+                return schema['id']
+
+        for schema in self._session.schemas:
+            if schema['id'].lower() == entity_type:
+                    return schema['id']
+
+        raise ValueError(
+            'Unable to translate entity type.'
+        )
+
+    def _launch(self, event):
+        args = self._translate_event(
+            self._session, event
+        )
+
+        interface = self._interface(
+            self._session, *args
+        )
+
+        if interface:
+            return interface
+
+        response = self.launch(
+            self._session, *args
+        )
+
+        return self._handle_result(
+            self._session, response, *args
+        )
+
+    def launch(self, session, uid, entities, source, values, event):
+        '''Callback method for the custom action.
+
+        return either a bool (True if successful or False if the action failed)
+        or a dictionary with they keys `message` and `success`, the message
+        should be a string and will be displayed as feedback to the user,
+        success should be a bool, True if successful or False if the action
+        failed.
+
+        *session* is a `ftrack_api.Session` instance
+
+        *uid* is the unique identifier for the event
+
+        *entities* is a list of tuples each containing the entity type and the
+        entity id.
+        If the entity is a hierarchical you will always get the entity
+        type TypedContext, once retrieved through a get operation you
+        will have the "real" entity type ie. example Shot, Sequence
+        or Asset Build.
+
+        *source* dictionary containing information about the source of the
+        event, application id, current user etc.
+
+        *values* is a dictionary containing potential user settings
+        from previous runs.
+
+        *event* the unmodified original event
+
+        '''
+        user = session.query(
+            'User where username is "{0}"'.format(os.environ["LOGNAME"])
+        ).one()
+        job = session.create(
+            'Job',
+            {
+                'user': user,
+                'status': 'running',
+                'data': json.dumps({
+                    'description': 'Create Structure: Scanning for data.'
+                })
+            }
+        )
+        # Commit to feedback to user about running job.
+        session.commit()
+
+        try:
```

Comment:
Given this is the only unique 'directory creation' code for this action, wouldn't it be better to create a generic action which takes a callback/topic/description/data as arguments, so we can reuse it?